shiny/types.py

```diff
@@ -31,24 +31,30 @@
 from htmltools import TagChild
 
 from ._docstring import add_example
-from ._typing_extensions import NotRequired, TypedDict
+from ._typing_extensions import NotRequired, TypedDict, TypeIs
 
 if TYPE_CHECKING:
     from matplotlib.figure import Figure
 
+T = TypeVar("T")
+
 
 # Sentinel value - indicates a missing value in a function call.
 class MISSING_TYPE:
     pass
 
 
 MISSING: MISSING_TYPE = MISSING_TYPE()
+DEPRECATED: MISSING_TYPE = MISSING_TYPE()  # A MISSING that communicates deprecation
+MaybeMissing = Union[T, MISSING_TYPE]
```

Comment:
I find `bg: MaybeMissing[str | None] = MISSING` a bit more readable than `bg: str | None | MISSING_TYPE = MISSING`, so this just provides a bit of sugar around the common pattern.

Reply:
This does seem nice, but also one of those things that would ideally be done across the entire codebase in one dedicated PR

Reply:
And to help clarify, I'm not saying I want it removed right now, but if we commit to it here, let's commit to it wherever it makes sense?

Reply:
I agree but I'm not up for doing the bigger refactor now. I opened https://github.com/posit-dev/py-shiny/issues/1873 and I'll take these changes out of this PR.